services/wallet/price.go

```diff
@@ -0,0 +1,82 @@
+package wallet
+
+import (
+	"database/sql"
+	"fmt"
+	"strconv"
+)
+
+type PricesPerTokenAndCurrency = map[string]map[string]float64
+
+type PriceManager struct {
+	db *sql.DB
+}
+
+func NewPriceManager(db *sql.DB) *PriceManager {
+	return &PriceManager{db: db}
+}
+
+func (pm *PriceManager) FetchPrices(symbols []string, currencies []string) (PricesPerTokenAndCurrency, error) {
+	result, err := fetchCryptoComparePrices(symbols, currencies)
+	if err != nil {
+		return result, err
+	}
+	err = pm.updatePriceCache(result)
+	return result, err
+}
+
+func (pm *PriceManager) updatePriceCache(prices PricesPerTokenAndCurrency) error {
+	insert, err := pm.db.Prepare(`INSERT OR REPLACE INTO price_cache 
+		(token,	currency, price)
+		VALUES
+		(?, ?, ?)`)
+	if err != nil {
+		return err
+	}
+
+	for token, pricesPerCurrency := range prices {
+		for currency, price := range pricesPerCurrency {
+			_, err = insert.Exec(token, currency, fmt.Sprintf("%f", price))
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func getCachedPricesFromDBRows(rows *sql.Rows) (PricesPerTokenAndCurrency, error) {
+	prices := make(PricesPerTokenAndCurrency)
+	for rows.Next() {
+		var token string
+		var currency string
+		var priceStr string
+		if err := rows.Scan(&token, &currency, &priceStr); err != nil {
+			return prices, err
+		}
+
+		_, present := prices[token]
+		if !present {
+			prices[token] = map[string]float64{}
+		}
+
+		price, err := strconv.ParseFloat(priceStr, 64)
```

Comment:
If the data type of the price column is `REAL`, there is no need for this conversion; `Scan` will do it instead.